src/frame.c

```diff
@@ -472,7 +472,11 @@ void frame_process(frame_t *st, size_t length)
         hef_t hef = {0};
 
         if (!fix_header(st, st->buffer + offset))
+        {
+            if (offset == 0)
+                st->input->sync_state = SYNC_STATE_NONE;
```

Comment:
Do you want to always do this or after N (e.g. 2 or 3) failures? I'm just imagining a case where there is some periodic interference or weak signal.

Reply:
I'll do some experimenting with that. I've done some testing, and the current design can result in significantly fewer audio packets being decoded when signal conditions are marginal.

Reply:
@awesie It turns out the difference in the number of audio packets being decoded was only because some stations are broadcasting P3 PDUs that don't seem to have valid audio packets in them. This was causing unnecessary sync resets. I've added a second commit which ensures that a reset only occurs when a P1 PDU doesn't have any decodable audio packets. After this change, there's essentially no difference in performance.

Reply:
This looks okay. A quick nitpick: maybe add a comment or something so that it is obvious that "length == MAX_PDU_LEN" is asserting that we are processing a P1 frame.

Reply:
Done.